packages/project-access/src/file/file-access.ts

```diff
@@ -66,3 +68,42 @@ export async function fileExists(path: string, memFs?: Editor): Promise<boolean>
         return false;
     }
 }
+
+/**
+ * Updates package.json file asynchronously by keeping the previous indentation.
+ *
+ * @param path - path to file
+ * @param packageJson - updated package.json file content
+ * @param memFs - optional mem-fs-editor instance
+ */
+export async function updatePackageJSON(path: string, packageJson: Package, memFs?: Editor): Promise<void> {
+    await updateJSON(path, packageJson, memFs);
+}
+
+/**
+ * Updates manifest.json file asynchronously by keeping the previous indentation.
+ *
+ * @param path - path to file
+ * @param manifest - updated manifest.json file content
+ * @param memFs - optional mem-fs-editor instance
+ */
+export async function updateManifestJSON(path: string, manifest: Manifest, memFs?: Editor): Promise<void> {
+    await updateJSON(path, manifest, memFs);
+}
+
+/**
+ * Updates JSON file asynchronously by keeping the indentation from previous content with new content for given path.
+ *
+ * @param path - path to file
+ * @param content - updated JSON file content
+ * @param memFs - optional mem-fs-editor instance
+ */
+async function updateJSON(path: string, content: object, memFs?: Editor): Promise<void> {
+    // read old contents and indentation of the JSON file
+    const oldContentText = await readFile(path, memFs);
+    const oldContentJson = parseJsonError(oldContentText);
+    const indent = Symbol.for('indent');
+    // prepare new JSON file content with previous indentation
+    const result = JSON.stringify(content, null, oldContentJson[indent]) + '\n';
```

Comment:
@Klaus-Keller 
should we consider `\r\n` line endings?
I think example in https://github.com/npm/json-parse-even-better-errors contains it